src/query/storage/fanout/storage.go

```diff
@@ -264,7 +263,20 @@ func (s *fanoutStorage) SearchSeries(
 		}
 
 		metadata = metadata.CombineMetadata(results.Metadata)
-		metrics = append(metrics, results.Metrics...)
+		for _, metric := range results.Metrics {
+			id := string(metric.ID)
+			if existing, found := metricMap[id]; found {
+				existing.Tags = existing.Tags.AddDedupedTags(metric.Tags.Tags)
```

Comment:
Interesting.  

What we used to do was make these timeseries look like different metrics even though they had the same ID.  i.e. you would just get two back in the series list and each with a different set of tags.  

It meant you never could use a map when collecting time series uniquely (since IDs could collide), but had the nice side effect of if you had the same time series with slightly different tags but they shared the same ID for some reason you could more easily debug that situation (since you could see them both in the results set).

Reply:
For now this is probably fine.  Not sure really what we should do long term though.

Either way we should probably standardize between what happens between a Search vs a Fetch though yeah?  

Seems like for Fetch it would actually return two different series (even if two IDs did collide)?